### PR TITLE
Fix for broken MBL

### PR DESCRIPTION
**Pull request by ghost**
LCD call was waiting forever, causing the system to hang.

### CODE REVIEW

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -1298,10 +1298,10 @@ void kill_screen(const char* lcd_msg) {
         #endif
       ;
       if (no_reentrance) return;
```
Comment (manianac) [inline Marlin/ultralcd.cpp]: Since we are no longer setting `no_reentrance` with `lcd_wait_for_homing`, I would remove this line.

Comment (ghost) [reply Marlin/ultralcd.cpp]: Done.

**Review by manianac — APPROVED** [review]

**Review by shacal — APPROVED** [review]

**Comment by ghost** [comment]
Can we ditch the whole `lcd_wait_for_homing()` function? I cannot find any other calls to it.

**Comment by thinkyhead** [comment]
> Can we ditch the whole lcd_wait_for_homing() function?

No. It's okay that it's only used once, currently. And I think the proper fix is actually this:
```diff
  inline void lcd_wait_for_homing() {
    no_reentrance = true;
    while (!axis_homed[X_AXIS] || !axis_homed[Y_AXIS] || !axis_homed[Z_AXIS]) idle();
-   no_reentrance = true;
+   no_reentrance = false;
  }
```

**Comment by thinkyhead** [comment]
Should be fixed now, with https://github.com/MarlinFirmware/Marlin/commit/fb3c30e24cf2ffc4b0e3109c1fd24f2a8dc4b84d, as above. Please confirm.

**Comment by ghost** [comment]
First off, welcome back, @thinkyhead!

I can try it this evening.

**Comment by thinkyhead** [comment]
Good to be back. I hope to be around a bit more. Still much to do in this move to Texas….

**Comment by ghost** [comment]
That commit didn't work. Details are on your commit.

**Comment by WheresWaldo** [comment]
Is there any movement on this PR. I had a bunch of reports from the Robo3D users that have migrated to 1.1.0 RCBugFix that are reporting MESH freezing during implementation. All with code from December 31, 2016. 

I was trying to maintain configuration files for the Robo R1 printer and everyone now has gone back to RC8 since the anomalous behavior is not present.

**Comment by manianac** [comment]
Awaiting @Thinkyhead s return, until then you'll have to experiment with your own PR or code regression.

**Comment by Sparkman61** [comment]
For what its worth, the change posted by @Thinkyhead seemed to correct the freezing problem. At least on my printer.

**Comment by WheresWaldo** [comment]
I did not know Scott was away. I will wait for his return.

**Comment by thinkyhead** [comment]
I've made the move to Austin, but my 3D printers are still in the mail. Once they arrive (in 2-3 days) I will be able to troubleshoot more. This PR won't be the solution, so I'm closing it.

**Pull request closed without merging**